controllers/managedosversionchannel_controller.go

```diff
@@ -225,15 +229,15 @@ func (r *ManagedOSVersionChannelReconciler) handleSyncPod(ctx context.Context, p
 			Status:  metav1.ConditionFalse,
 			Message: "ongoing channel synchronization",
 		})
-		return ctrl.Result{RequeueAfter: r.minTimeBetweenSyncs / 2}, nil
+		return ctrl.Result{RequeueAfter: r.minTimeBetweenSyncs / 2}
 	case corev1.PodSucceeded:
 		data, err = r.syncerProvider.ReadPodLogs(ctx, r.kcl, pod, displayContainer)
 		if err != nil {
-			return ctrl.Result{}, err
+			return ctrl.Result{RequeueAfter: interval}
```

Comment:
On error we just requeue to the next synchronization interval. The status (failed or not) is tracked in logs and in ready condition.